AnkiDroid/src/main/java/com/ichi2/async/coroutines/CoroutineJob.kt

```
@@ -0,0 +1,158 @@
+/****************************************************************************************
+ * Copyright (c) 2022 Divyansh Kushwaha <kushwaha.divyansh.dxn@gmail.com>               *
+ *                                                                                      *
+ * This program is free software; you can redistribute it and/or modify it under        *
+ * the terms of the GNU General Public License as published by the Free Software        *
+ * Foundation; either version 3 of the License, or (at your option) any later           *
+ * version.                                                                             *
+ *                                                                                      *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY      *
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A      *
+ * PARTICULAR PURPOSE. See the GNU General Public License for more details.             *
+ *                                                                                      *
+ * You should have received a copy of the GNU General Public License along with         *
+ * this program.  If not, see <http://www.gnu.org/licenses/>.                           *
+ ****************************************************************************************/
+
+package com.ichi2.async.coroutines
+
+import android.annotation.SuppressLint
+import kotlinx.coroutines.*
+import timber.log.Timber
+
+/**
+ * This class is equivalent to AsyncTask, it keeps the functionality while using Kotlin-Coroutines beneath
+ */
+abstract class CoroutineJob<Params, Progress, Result>(
```

Would be good to include coroutine exception handling in this class.

---

Could you explain where should I add exception handling, because as far as I understand it should be caller's responsibility to check for error as we don't know what code goes inside doInBackground, might be network call, database query etc, so catching it here would silence the exception.

---

It's okay if it's being handled at the caller side.